plugin-server/functional_tests/api.ts

```diff
@@ -145,16 +146,16 @@ export const reloadDictionaries = async () => {
 
 export const fetchEvents = async (teamId: number, uuid?: string) => {
     const queryResult = (await clickHouseClient.querying(`
-        SELECT
-            *,
-            dictGetOrDefault(
-                person_overrides_dict,
-                'override_person_id',
-                (${teamId}, events.person_id),
-                events.person_id
-            ) as person_id
-        FROM events
-        WHERE team_id = ${teamId} ${uuid ? `AND uuid = '${uuid}'` : ``} ORDER BY timestamp ASC
+        SELECT *,
+               if(notEmpty(overrides.person_id), overrides.person_id, e.person_id) as person_id
+        FROM events e
+                 LEFT OUTER JOIN
+             (SELECT override_person_id as person_id,
+                     old_person_id
+              FROM person_overrides
+              WHERE team_id = ${teamId}) AS overrides ON e.person_id = overrides.old_person_id
+        WHERE team_id = ${teamId} ${uuid ? `AND uuid = '${uuid}'` : ``}
+        ORDER BY timestamp ASC
```

Comment:
easier to see example - updated the code to have custom event 4 to be the last one
```
SELECT
    timestamp,
    event
FROM events
WHERE team_id = 156

Query id: e6138ae7-4114-46a6-a872-ae65dc092af3

┌──────────────────timestamp─┬─event──────────────┐
│ 2023-04-26 13:10:31.905000 │ $merge_dangerously │
└────────────────────────────┴────────────────────┘
┌──────────────────timestamp─┬─event──────────┐
│ 2023-04-26 13:10:30.839000 │ $identify      │
│ 2023-04-26 13:10:30.832000 │ $identify      │
│ 2023-04-26 13:10:29.781000 │ custom event   │
│ 2023-04-26 13:10:29.789000 │ custom event 2 │
│ 2023-04-26 13:10:29.796000 │ custom event 3 │
│ 2023-04-26 13:10:29.802000 │ custom event 4 │
└────────────────────────────┴────────────────┘

7 rows in set. Elapsed: 0.003 sec. 

5fb33ba45162 :) SELECT e.event,
                               if(notEmpty(overrides.person_id), overrides.person_id, e.person_id) as person_id
                        FROM events e
                                 LEFT OUTER JOIN
                             (SELECT override_person_id as person_id,
                                     old_person_id
                              FROM person_overrides
                              WHERE team_id = 156) AS overrides ON e.person_id = overrides.old_person_id
                        WHERE team_id = 156
                        ORDER BY timestamp ASC

SELECT
    e.event,
    if(notEmpty(overrides.person_id), overrides.person_id, e.person_id) AS person_id
FROM events AS e
LEFT JOIN
(
    SELECT
        override_person_id AS person_id,
        old_person_id
    FROM person_overrides
    WHERE team_id = 156
) AS overrides ON e.person_id = overrides.old_person_id
WHERE team_id = 156
ORDER BY timestamp ASC

Query id: 5901126f-67c1-47c6-a997-bc81f37db01b

┌─event──────────┬─person_id────────────────────────────┐
│ custom event   │ 0187bdaf-d49d-0000-89ad-5bb7088a3c52 │
│ custom event 2 │ 0187bdaf-d49d-0000-89ad-5bb7088a3c52 │
│ custom event 3 │ 0187bdaf-d49d-0000-89ad-5bb7088a3c52 │
│ custom event 4 │ 0187bdaf-d4cf-0001-820b-2dfe986fc342 │
│ custom event 4 │ 0187bdaf-d49d-0000-89ad-5bb7088a3c52 │
│ $identify      │ 0187bdaf-d49d-0000-89ad-5bb7088a3c52 │
│ $identify      │ 0187bdaf-d49d-0000-89ad-5bb7088a3c52 │
└────────────────┴──────────────────────────────────────┘
┌─event──────────────┬─person_id────────────────────────────┐
│ $merge_dangerously │ 0187bdaf-d49d-0000-89ad-5bb7088a3c52 │
└────────────────────┴──────────────────────────────────────┘

8 rows in set. Elapsed: 0.007 sec. 
```

Reply:
This happens because of we're not using the latest value from the overrides table

```
5fb33ba45162 :) select * from person_overrides where team_id = 156

SELECT *
FROM person_overrides
WHERE team_id = 156

Query id: 5beb5aea-fccb-451c-849d-1a51c7f2f4d5

┌─team_id─┬─old_person_id────────────────────────┬─override_person_id───────────────────┬──────────────────merged_at─┬───────────────oldest_event─┬─────────────────created_at─┬─version─┐
│     156 │ 0187bdaf-d4cf-0001-820b-2dfe986fc342 │ 0187bdaf-d49d-0000-89ad-5bb7088a3c52 │ 2023-04-26 13:10:31.926000 │ 2023-04-26 13:10:29.781000 │ 2023-04-26 13:10:31.000000 │       0 │
│     156 │ 0187bdaf-d4d0-0000-fc4a-e6e9758156c5 │ 0187bdaf-d49d-0000-89ad-5bb7088a3c52 │ 2023-04-26 13:10:31.926000 │ 2023-04-26 13:10:29.796000 │ 2023-04-26 13:10:31.000000 │       1 │
└─────────┴──────────────────────────────────────┴──────────────────────────────────────┴────────────────────────────┴────────────────────────────┴────────────────────────────┴─────────┘
┌─team_id─┬─old_person_id────────────────────────┬─override_person_id───────────────────┬──────────────────merged_at─┬───────────────oldest_event─┬─────────────────created_at─┬─version─┐
│     156 │ 0187bdaf-d4cf-0000-d224-3419cdcb76ac │ 0187bdaf-d49d-0000-89ad-5bb7088a3c52 │ 2023-04-26 13:10:30.853000 │ 2023-04-26 13:10:29.781000 │ 2023-04-26 13:10:30.000000 │       0 │
│     156 │ 0187bdaf-d4d0-0000-fc4a-e6e9758156c5 │ 0187bdaf-d4cf-0001-820b-2dfe986fc342 │ 2023-04-26 13:10:30.905000 │ 2023-04-26 13:10:29.796000 │ 2023-04-26 13:10:30.000000 │       0 │
└─────────┴──────────────────────────────────────┴──────────────────────────────────────┴────────────────────────────┴────────────────────────────┴────────────────────────────┴─────────┘

4 rows in set. Elapsed: 0.003 sec. 
```